File: combinatorics/resource/ST/project_5/bai_toan_4/mutil_graph/al_to_am.py
def al_to_am(adj_list):
    n = len(adj_list)
    matrix = [[0]*n for _ in range(n)]  # Khoi tao ma tran 0

    for i in range(n):
        for j in adj_list[i]:
            matrix[i][j] += 1  # Tang dem so canh tu i den j

    return matrix

# Nhap tu ban phim
n = int(input("Nhap so dinh: "))
adj_list = []

print("Nhap danh sach ke:")
for i in range(n):
    row = list(map(int, input(f"Ke cua dinh {i}: ").split()))
    adj_list.append(row)

# Chuyen doi va in ket qua
matrix = al_to_am(adj_list)
print("Ma tran ke:")
for row in matrix:
    print(" ".join(map(str, row)))

File: combinatorics/resource/ST/project_5/bai_toan_4/mutil_graph/al_to_am.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

vector<vector<int>> alToAm(const vector<vector<int>>& adjList) {
    int n = adjList.size();
    vector<vector<int>> matrix(n, vector<int>(n, 0)); // Khoi tao ma tran 0

    for (int i = 0; i < n; ++i) {
        for (int j : adjList[i]) {
            matrix[i][j]++; // Tang dem moi lan co canh
        }
    }

    return matrix;
}

int main() {
    int n;
    cout << "Nhap so dinh: ";
    cin >> n;

    vector<vector<int>> adjList(n);
    cout << "Nhap danh sach ke (cac so cach nhau bang dau cach):\n";
    cin.ignore(); // Xoa bo nho dem

    for (int i = 0; i < n; ++i) {
        cout << "Ke cua dinh " << i << ": ";
        string line;
        getline(cin, line);
        istringstream iss(line);
        int x;
        while (iss >> x)
            adjList[i].push_back(x);
    }

    // Chuyen doi va in ket qua
    vector<vector<int>> matrix = alToAm(adjList);
    cout << "Ma tran ke:\n";
    for (const auto& row : matrix) {
        for (int val : row)
            cout << val << " ";
        cout << endl;
    }

    return 0;
}